main/src/io/github/iltotore/iron/constraint/any.scala

```diff
@@ -164,9 +165,26 @@ object any:
      */
     given right[C1, C2, C3](using C1 ==> Not[C2], C1 ==> C3): (C1 ==> Xor[C2, C3]) = Implication()
 
-  object StrictEqual:
-    inline given [A, V]: Constraint[A, StrictEqual[V]] with
+  object StrictEqual extends StrictEqualLowPriority:
 
-      override inline def test(value: A): Boolean = value == constValue[V]
+    inline given [V <: NumConstant]: StrictEqualConstraint[BigDecimal, V] with
+      override inline def test(value: BigDecimal): Boolean = value == doubleValue[V]
+
+    inline given [V <: IntConstant]: StrictEqualConstraint[BigInt, V] with
+      override inline def test(value: BigInt): Boolean = value == longValue[V]
+
+    inline given jBigDecimal[V <: NumConstant]: StrictEqualConstraint[java.math.BigDecimal, V] with
+      override inline def test(value: java.math.BigDecimal): Boolean =
+        value == java.math.BigDecimal(stringValue[V])
+
+    inline given jBigInteger[V <: IntConstant]: StrictEqualConstraint[java.math.BigInteger, V] with
+      override inline def test(value: java.math.BigInteger): Boolean =
+        value == java.math.BigInteger(stringValue[V])
 
+  sealed trait StrictEqualLowPriority:
```

Comment:
Is it necessary? There should be no conflict since Scala takes the most specific given instance when resolving implicits.

Reply:
Fixed